### PR TITLE
Wait for model deployment to complete before deleting old artefacts

### PR DESCRIPTION
**Pull request by barrucadu**
Example: https://cd.gds-reliability.engineering/teams/govuk-tools/pipelines/search-learn-to-rank/jobs/staging-deploy/builds/6

---

The deploy method has a parameter called "wait":

> wait (bool) – Whether the call should wait until the deployment of
> this model completes (default: True).

We're not setting it to false, but this script is proceeding to delete
the old endpoint config and model before the endpoint has updated to
the new one.

### CODE REVIEW

File: ltr/concourse/deploy.py [hunk 1]
```diff
@@ -43,6 +45,19 @@
     update_endpoint=True,
 )
 
+# wait for the deployment to complete
+endpoint_status = "Updating"
+while endpoint_status == "Updating":
```
Comment (bilbof) [inline ltr/concourse/deploy.py:50]: Do you think checking if `endpoint_status != OK` (replacing OK with whatever the OK status is) would help? I'm wondering if there are other statuses that aren't OK during which we'd also want to wait.

**Review by bilbof — APPROVED** [review]
LGTM!

File: ltr/concourse/deploy.py (hunk 1 above)
Comment (barrucadu) [reply ltr/concourse/deploy.py:50]: I think we want to bail out when it's not `Updating`.  There's a list of them here: https://docs.aws.amazon.com/cli/latest/reference/sagemaker/describe-endpoint.html

`Creating` won't happen (outside the bootstrap job), `SystemUpdating` means AWS maintenance is happening and we don't know how long it'll take (and if we're checking for that we should do it before even trying to deploy the model), and `RollingBack` will get back to the old working configuration successfully if we don't delete it.

Comment (bilbof) [reply ltr/concourse/deploy.py:50]: OK, thanks for that detail!

**Pull request merged**